core/Test/Tasty/Core.hs

```diff
@@ -65,8 +68,22 @@ data Result = Result
     -- @FAIL@.
   , resultTime :: Time
     -- ^ How long it took to run the test, in seconds.
+  , printResultDetails :: ResultDetailsPrinter
+    -- ^ Optional detailed result printing action.
+    --
+    -- 'printResultDetails' is used by providers that need to have fine grained
+    -- control about the printed results. Especially when it comes to colorization.
   }
-  deriving Show
+
+instance Show Result where
```

Comment:
This instance exists as the `ResultDetailsPrinter` itself has no `Show` instance.

An alterative would be to wrap it into a type with explicit "dummy" show instance. I'm not sure which one you'd prefer.